### PR TITLE
[ODS-5355] Eliminate EdOrg cache and supporting components and tests

### PR DESCRIPTION
**Pull request by gmcelhanon**
See also: https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/pull/410

### CODE REVIEW

File: Application/EdFi.Ods.Api/Container/Modules/SqlServerSpecificModule.cs [hunk 1]
```diff
@@ -28,10 +28,6 @@ public override void ApplyConfigurationSpecificRegistrations(ContainerBuilder bu
                 .As<IParameterListSetter>()
                 .SingleInstance();
 
-            builder.RegisterType<SqlServerAuthorizationSegmentSqlProvider>()
```
Comment (AxelMarquez) [inline Application/EdFi.Ods.Api/Container/Modules/SqlServerSpecificModule.cs:31]: The next using is no longer needed:
`using EdFi.Ods.Api.Security.Authorization;`

File: Application/EdFi.Ods.Api/Container/Modules/PostgresSpecificModule.cs [hunk 2]
```diff
@@ -28,10 +28,6 @@ public override void ApplyConfigurationSpecificRegistrations(ContainerBuilder bu
                 .As<IParameterListSetter>()
                 .SingleInstance();
 
-            builder.RegisterType<PostgresAuthorizationSegmentSqlProvider>()
```
Comment (AxelMarquez) [inline Application/EdFi.Ods.Api/Container/Modules/PostgresSpecificModule.cs:31]: The next using is no longer needed:
`using EdFi.Ods.Api.Security.Authorization;`

File: Application/EdFi.Ods.Api/Security/Authorization/EducationOrganizationIdNamesProvider.cs [hunk 3]
```diff
@@ -13,86 +13,25 @@
...
     public class EducationOrganizationIdNamesProvider : IEducationOrganizationIdNamesProvider
     {
-        private readonly Lazy<BidirectionalGraph<string, Edge<string>>> _edOrgIdGraph;
         private readonly Lazy<string[]> _concreteEducationOrganizationIdNames;
-        private readonly Lazy<HashSet<(string, string)>> _accessibleSegments;
         private readonly Lazy<List<string>> _sortedEducationOrganizationIdNames; 
 
         public EducationOrganizationIdNamesProvider(IDomainModelProvider domainModelProvider)
         {
-            _edOrgIdGraph = new Lazy<BidirectionalGraph<string, Edge<string>>>(
```
Comment (AxelMarquez) [inline Application/EdFi.Ods.Api/Security/Authorization/EducationOrganizationIdNamesProvider.cs:25]: The next usings are no longer needed:
```
using EdFi.Ods.Common.Models.Domain;
using QuickGraph;
```

File: Application/EdFi.Ods.Api/Security/Authorization/Repositories/AggregateRootCriteriaProviderDecoratorBase.cs [hunk 4]
```diff
@@ -31,31 +30,17 @@ public abstract class AggregateRootCriteriaProviderAuthorizationDecoratorBase<TE
         private readonly IAggregateRootCriteriaProvider<TEntity> _decoratedInstance;
         private readonly IAuthorizationFilterContextProvider _authorizationFilterContextProvider;
         private readonly IAuthorizationFilterDefinitionProvider _authorizationFilterDefinitionProvider;
-        private readonly IEducationOrganizationIdNamesProvider _educationOrganizationIdNamesProvider;
-        private readonly Lazy<List<string>> _sortedEducationOrganizationIdNames;
 
         private readonly ILog _logger;
```
Comment (AxelMarquez) [inline Application/EdFi.Ods.Api/Security/Authorization/Repositories/AggregateRootCriteriaProviderDecoratorBase.cs:34]: `_logger` is assigned but never used, can we remove it?

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/SqlServerViewBasedSingleItemAuthorizationQuerySupport.cs [hunk 5]
```diff
@@ -34,7 +34,7 @@ public string GetItemExistenceCheckSql(ViewBasedAuthorizationFilterDefinition fi
     public void ApplyClaimsParametersToCommand(DbCommand cmd, EdFiAuthorizationContext authorizationContext)
     {
         // No parameters needed if less than 2,000 EdOrgIds present
-        if (authorizationContext.ApiKeyContext.EducationOrganizationIds.Count() < SqlServerParameterCountThreshold)
+        if (authorizationContext.ApiKeyContext.EducationOrganizationIds.Length < SqlServerParameterCountThreshold)
```
Comment (AxelMarquez) [inline Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/SqlServerViewBasedSingleItemAuthorizationQuerySupport.cs:37]: The next using is no longer needed:
`using System.Linq;`

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsWithStudentsOnlyAuthorizationStrategy.cs [hunk 6]
```diff
@@ -4,6 +4,7 @@
 // See the LICENSE and NOTICES files in the project root for more information.
 
 using System;
+using System.Collections.Generic;
```
Comment (AxelMarquez) [inline Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsWithStudentsOnlyAuthorizationStrategy.cs:7]: The next usings are no longer needed:
```
using System;
using EdFi.Ods.Api.Security.Authorization;
```

File: Application/EdFi.Ods.Api/Security/Container/Modules/SecurityPersistenceModule.cs [hunk 7]
```diff
@@ -61,18 +61,6 @@ protected override void Load(ContainerBuilder builder)
                 builder.RegisterDecorator(decoratorRegistration.Value, decoratorRegistration.Key);
             }
 
-            builder.RegisterType<EducationOrganizationCache>()
```
Comment (AxelMarquez) [inline Application/EdFi.Ods.Api/Security/Container/Modules/SecurityPersistenceModule.cs:64]: The next usings are no longer needed:
```
using Autofac.Core;
using EdFi.Ods.Common.Caching;
using EdFi.Ods.Api.Security.Authorization;
```

**Review by AxelMarquez — DISMISSED** [review]
Looks good to me; left a couple of minor comments related to unused usings.

File: Application/EdFi.Ods.Api/Container/Modules/SqlServerSpecificModule.cs (hunk 1 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Container/Modules/SqlServerSpecificModule.cs:31]: Done.

File: Application/EdFi.Ods.Api/Container/Modules/PostgresSpecificModule.cs (hunk 2 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Container/Modules/PostgresSpecificModule.cs:31]: Done.

File: Application/EdFi.Ods.Api/Security/Authorization/EducationOrganizationIdNamesProvider.cs (hunk 3 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Security/Authorization/EducationOrganizationIdNamesProvider.cs:25]: Done.

File: Application/EdFi.Ods.Api/Security/Authorization/Repositories/AggregateRootCriteriaProviderDecoratorBase.cs (hunk 4 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Security/Authorization/Repositories/AggregateRootCriteriaProviderDecoratorBase.cs:34]: Done.

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsWithStudentsOnlyAuthorizationStrategy.cs (hunk 6 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsWithStudentsOnlyAuthorizationStrategy.cs:7]: Done.

File: Application/EdFi.Ods.Api/Security/Container/Modules/SecurityPersistenceModule.cs (hunk 7 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Security/Container/Modules/SecurityPersistenceModule.cs:64]: Done.

**Review by bptillman — APPROVED** [review]

**Pull request merged**